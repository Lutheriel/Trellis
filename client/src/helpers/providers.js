import React from 'react'
import { Provider } from 'react-redux'
import store from '../store/configureStore'
import { DragDropContext, Droppable } from 'react-beautiful-dnd'

export const TestProvider = ({ children }) => {
  return (
    <Provider store={store}>
      <DragDropContext onDragEnd={() => {}}>{children}</DragDropContext>
    </Provider>
  )
}
