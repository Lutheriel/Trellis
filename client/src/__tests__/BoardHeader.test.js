import React from 'react'
import { render } from '@testing-library/react'
import BoardHeader from '../components/BoardHeader'
import renderer from 'react-test-renderer'
import { TestProvider } from '../helpers/providers'

it('should display correctly BoardHeader', async () => {
  const { getByText, getByTestId } = render(
    <TestProvider>
      <BoardHeader
        title= 'table1'
        showEditable={() => {}}
      />
    </TestProvider>
  )

  const table = getByText('table1')
  expect(window.getComputedStyle(table).display).toEqual('block')
});

it('should check snapshot for BoardHeader', () => {
  const tree = renderer
    .create(
      <TestProvider>
        <BoardHeader
          title='table1'
          showEditable={() => {}}
        />
      </TestProvider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
