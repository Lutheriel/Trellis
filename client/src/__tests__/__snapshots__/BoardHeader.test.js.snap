// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should check snapshot for BoardHeader 1`] = `
<div
  className="makeStyles-container-3"
>
  <div
    className="makeStyles-title-4"
    onClick={[Function]}
  >
    table1
  </div>
</div>
`;
